src/main/java/picard/sam/util/SamComparison.java

```diff
@@ -155,12 +168,18 @@ private boolean compareAlignmentsAndCatalogDuplicateMarkingDifferences() {
         if (!compareValues(leftReader.getFileHeader().getSortOrder(), rightReader.getFileHeader().getSortOrder(),
                 "Sort Order")) {
             System.out.println("Cannot compare alignments if sort orders differ.");
+            if (samComparisonArgumentCollection.OUTPUT_MQ_CONCORDANCE != null) {
+                throw new PicardException("Cannot generate mapping quality concordance histogram because the two files cannot be compared (see reason above).");
```

Comment:
don't throw an exception.  we want the tool to run successfully and return that the SAM files are different in this case.